crates/astria-sequencer/src/app.rs

```diff
@@ -361,7 +361,7 @@ impl App {
     /// successfully.
     ///
     /// Note that `begin_block` is now called *after* transaction execution.
-    #[instrument(name = "App::deliver_tx", skip(self))]
+    #[instrument(name = "App::deliver_tx", skip_all, fields(tx_hash = hex::encode(signed_tx.hash())))]
```

Comment:
Several notes on this:

1. I want to get away from `hex::encode` because it has an unnecessary allocation. Can you write a newtype wrapper like`DisplayHex<'a, T>(&'a T)` with `impl<'a, T> std::fmt::Display for DisplayHex<'a, T> where T: AsRef<[u8>]` that walks the bytes and writes the individual bytes to the formatter like here? https://github.com/astriaorg/astria/blob/04cfccb5109df7aa953d2767b7c26a5605c0931e/crates/astria-core/src/sequencer/v1alpha1/mod.rs#L97-L104
2. I think it would be nice to have the hash of both the signed and the unsigned transactions here, and the public key. Not sure if this is overkill, but IMO very useful. The reason to have the hash of the unsigned transactions is that on the composer side we only get the hash of the signed transaction immediately before sending. The signed tx hash also changes if the nonce is wrong, while the unsigned tx remains stable.
3. Also please use more verbose field names. I am thinking something like `hash.signed_transaction`, `hash.unsigned_transaction` and `sender`